modules/mocked_authentication/lib/mockdata/reader.rb

```diff
@@ -0,0 +1,67 @@
+# frozen_string_literal: true
+
+module MockedAuthentication
+  module Mockdata
+    class Reader
+      def self.find_credentials(credential_type:)
+        file_names = get_file_names(credential_type)
+        get_mocked_data(file_names)
+      end
+
+      class << self
+        private
+
+        def get_file_names(type)
+          file_directory = get_file_directory_for_type(type)
+          users = Dir.entries(file_directory).select { |file| validate_file_name(file) ? file : nil }
```

Comment:
After getting `file_directory`, I would use `Dir.glob`:

`Dir.glob("#{file_directory}*")`

That will do what you're doing here more succinctly